scripts/biomedical/diseasesAtJensenLab/scripts/format_disease_jensen_lab.py

```diff
@@ -0,0 +1,383 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Author: Suhana Bedi
+Date: 08/12/2023
+Edited By: Samantha Piekos
+Last Edited: 2/7/25
+Name: format_disease_jensen_lab
+Description: converts a three input .txt files from Diseases at
+Jensen Lab into output .csv files with formatted dcids, 
+NonCoding RNA types, and gene ensemblIDs.
+@file_input: input .txt files from Diseases at Jensen Lab 
+@file_output: formatted .csv files for Diseases at Jensen Lab
+"""
+
+# load environment
+import sys
+import numpy as np
+import pandas as pd
+import time
+
+
+# declare universal variables
+HGNC_DICT = {'HGNC:9982':'RFX1', 'HGNC:9979':'RFPL2'}
+LIST_ILLEGAL_CHARACTERS = ["'", "*" ">", "<", "@", "]", "[", "|", ":", ";" " "]
+DICT_COL_NAMES = {'experiments': 
+	[
+	'Id',
+	'Gene',
+	'Disease',
+	'Disease_Name',
+	'score-db',
+	'source-score',
+	'confidence'
+	],
+	'knowledge':
+	[
+	'Id',
+	'Gene',
+	'Disease',
+	'Disease_Name',
+	'score-db',
+	'evidence',
+	'confidence'
+	],
+	'textmining':
+	[
+	'Id',
+	'Gene',
+	'Disease',
+	'Disease_Name',
+	'z-score',
+	'confidence',
+	'url'
+	]
+}
+
+
+
+def filter_for_lowest_ICD10_level(df):
+    """Filters for the most specific ICD10 code for each gene."""
+
+    indices_to_drop = []
+    previous_ICD10 = ''
+    previous_gene = ''
+    previous_index = -1
+
+    for index, row in df.iterrows():
+        current_ICD10 = row['ICD10']
+        current_gene = row['Gene']
+
+        # Check if current ICD10 is more specific than previous for the same gene
+        if current_gene == previous_gene and current_ICD10.startswith(previous_ICD10):
+            indices_to_drop.append(previous_index)
+
+        # Update tracking variables
+        previous_ICD10 = current_ICD10
+        previous_gene = current_gene
+        previous_index = index
+
+    # drop rows with less specific ICD10 codes for a given gene
+    df.drop(indices_to_drop, axis=0, inplace=True)
+    
+    return df
+
+
+def fix_ICD10_formatting(df, col):
+	"""
+	Adds a missing decimal in proper position to more specific ICD10 codes 
+	that are missing them. If string in specified column is 10 characters or longer, 
+	add '.' in the 10th position.
+	"""
+	mask = df[col].str.len() >= 10
+	df.loc[mask, col] = df.loc[mask, col].str[:9] + '.' + df.loc[mask, col].str[9:]
+	return df
+
+
+def format_icd10_code_dcids(df):
+	"""Formats ICD10 code DCIDs, filtering for the most specific codes."""
+	df_doid = df.dropna(subset=['DOID'])    # Create a separate DataFrame for rows with DOID
+
+	# Filter out invalid ICD-10 codes (e.g., root, short codes like C1 or C2)
+	df['count'] = np.where(df['ICD10'] == df['ICD10'],df['ICD10'].str.split(':'),np.nan)
```

Comment:
wouldn't df['ICD10'] always equal df['ICD10']? & why are we setting df['count'] here and re-setting it in the next line?